### PR TITLE
feat(rpc): implement `MpoolSelect`

### LINKED ISSUE #4196: MpoolSelect

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Expose `select_messages`, which was hidden behind `#[cfg(test)]`.
- Test schema of `MpoolSelect` against Lotus.

## Reference issue to close (if applicable)

Closes #4196 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by lemmih** [comment]
The newly exposed code could use a lot of love.

File: src/message_pool/msg_chain.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2019-2024 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
-
+#![allow(clippy::indexing_slicing)]
```
Comment (aatifsyed) [inline src/message_pool/msg_chain.rs:3]: I'd rather see documented unwraps

File: src/message_pool/msgpool/selection.rs [hunk 2]
```diff
@@ -6,6 +6,7 @@
 //! `select_messages` API which selects an appropriate set of messages such that
 //! it optimizes miner reward and chain capacity. See <https://docs.filecoin.io/mine/lotus/message-pool/#message-selection> for more details
 
+#![allow(clippy::indexing_slicing)]
```
Comment (aatifsyed) [inline src/message_pool/msgpool/selection.rs:9]: As above

**Review by aatifsyed — APPROVED** [review]

File: src/rpc/methods/mpool.rs [hunk 3]
```diff
@@ -119,6 +120,29 @@ impl RpcMethod<1> for MpoolPending {
     }
 }
 
+/// Return `Vec` of pending messages for inclusion in the next block
+pub enum MpoolSelect {}
+impl RpcMethod<2> for MpoolSelect {
+    const NAME: &'static str = "Filecoin.MpoolSelect";
+    const PARAM_NAMES: [&'static str; 2] = ["tsk", "tq"];
```
Comment (hanabi1224) [inline src/rpc/methods/mpool.rs:127]: Should we use full words, `tipset_key` and, what is `tq`?

**Review by hanabi1224 — APPROVED** [review]

File: src/message_pool/msg_chain.rs (hunk 1 above)
Comment (lemmih) [reply src/message_pool/msg_chain.rs:3]: Me too. I opened a new tracking issue: https://github.com/ChainSafe/forest/issues/4229

File: src/rpc/methods/mpool.rs (hunk 3 above)
Comment (lemmih) [reply src/rpc/methods/mpool.rs:127]: Other commands also use 'tsk'. Not sure what `tq` means. I blatantly copied it from Lotus. Hopefully, we can fix it once we understand what these RPC methods should do.

**Pull request merged**